code/modules/reagents/machinery/chem_dispenser.dm

```diff
@@ -316,6 +321,21 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 			return
 
 	ui_interact(user)
+
+/obj/machinery/chem_dispenser/update_icon()
+
+	overlays.len = 0
+	var/image/overlay
+
+	if(istype(container, /obj/item/weapon/reagent_containers/glass/beaker/bluespace) || istype(container, /obj/item/weapon/reagent_containers/glass/beaker/noreact))
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay2")
+	else
+		overlay = image('icons/obj/chemical.dmi', src, "dispenser_overlay1")
+
+	if(beaker)
```

Comment:
I don't get the point of this var. Couldn't you just check if `container` is null?

Reply:
fixed, thanks